### ISSUE TITLE
Feature Request: Make "Restore In New Tab" default in History pane

### ISSUE THREAD

**Issue by jim256** [hankinsoft/SQLPro#400]
I use the history pane quite a bit. It's rare that I want to replace my current query with what I'm seeing (though it does happen). Much more commonly, I want to use the "Restore In New Tab" option.

What are your thoughts on either a) making that the default, and making having to click the dropdown necessary to get to "Replace All" or b) making both buttons separate / visible (this option may make more sense if #399 is implemented?) ?

**Comment by hankinsoft** [comment]
What about 'Cmd-T' (same as the new tab shortcut)? Then if you have the history panel open and you hit 'new tab', it just uses what you have visible?

**Comment by jim256** [comment]
@hankinsoft yeah, that seems solid.

Also, I recognize that I could just always open a new tab first. The issue is that sometimes I don't find what I'm looking for, so in those cases I just end up closing that new tab.

**Comment by jim256** [comment]
@hankinsoft could I also make a few other history window requests? Happy to add these all as one new issue, or to make one separate new issue each if one of those options works better for you.

- Make the cursor focus automatically be in the search box when the history window opens, since as things stand now, typing does nothing and if you want to search you need to tab or mouse click to the box
- Something to improve navigation once a search has been executed. Right now, after you type your search term and hit enter, it gives you the results. If you hit enter again, it executes the search again. If you hit the arrow keys, it moves the arrows within the search term box. If you Shift + Tab over to the arrow keys and try to use Enter or Space on them, it inserts the current query. The only way I've found so far is to use (Shift +) Tab after executing the search, and then to use the arrow keys. Honestly that works fine once you know, but may not be as intuitive for users who don't already know. Not sure of the best way to do this one. Maybe CMD + Arrow keys?

**Comment by hankinsoft** [comment]
Hows something like this?

![Screen Shot 2020-04-01 at 5 05 57 PM](https://user-images.githubusercontent.com/3160528/78181638-202e0200-743b-11ea-9cf5-c37a405309f0.png)

Actions are now cmd + arrows plus cmd + return as the default action (I felt I need to put some text for existing users who may not know why the arrows without cmd are no longer working).

Clicking the replace all dropdown displays the shortcut for replace in new tab.

![Screen Shot 2020-04-01 at 5 07 37 PM](https://user-images.githubusercontent.com/3160528/78181814-59667200-743b-11ea-8535-edb67404890c.png)

**Comment by jim256** [comment]
@hankinsoft holy smokes you are quick. That looks really good!!

**Comment by hankinsoft** [comment]
I made a few changes to the text, but otherwise the item is all ready for
the next build. Resizing is still being investigated (the out of box system
pop up can not be resized). No ETA on that as of yet.

**Comment by hankinsoft** [comment]
Should be sorted in the latest App Store version.

**Comment by jim256** [comment]
Love all these improvements! Really nice.

One other *minor* thing worth considering could be to have the cursor focus in the search box by default. Obviously, it's pretty quick to get there with 2 tabs. But since we now have a shortcut to move back and forth, there is probably little utility in having the focus be on the arrow:

![image](https://user-images.githubusercontent.com/15026225/78582127-0bae8880-77f2-11ea-93ba-1b04c079240e.png)

**Comment by hankinsoft** [comment]
Hmmm, its suppose to be focused (and is in my testing)... I'll have to
investigate further.

**Comment by hankinsoft** [comment]
Latest build (2020.38) is available which should sort this out. Please let me know if you get a chance. For this one the last remaining item was to default the search as highlighted which it should be as of this build.

**Comment by jim256** [comment]
@hankinsoft, seems to be taken care of! Love it.

**Comment by jim256** [comment]
@hankinsoft, I did a really poor job capturing the full request in this issue (keep making adjustments).

One tiny tweak I just noticed that would be nice / make sense is for this shortcut to still work even if you don't have a tab open yet. I recognize that that's where you select your connection, but it didn't feel intuitive to have to open a blank tab in order to search my history. If this is pursued, I'd propose that hitting restore when no tab is open simply creates a new tab with the query selected.

Minor, for sure.
